kola/tests/misc/nvidia.go

```diff
@@ -3,50 +3,239 @@ package misc
 import (
 	"bytes"
 	"fmt"
+	"strings"
 	"time"
 
+	"github.com/coreos/go-semver/semver"
 	"github.com/coreos/pkg/capnslog"
 	"github.com/flatcar/mantle/kola"
 	"github.com/flatcar/mantle/kola/cluster"
 	"github.com/flatcar/mantle/kola/register"
+	testsutil "github.com/flatcar/mantle/kola/tests/util"
+	"github.com/flatcar/mantle/platform"
+	"github.com/flatcar/mantle/platform/conf"
 	"github.com/flatcar/mantle/util"
 )
 
 const (
 	CmdTimeout = time.Second * 300
 )
 
+const nvidiaDriverVersionOverride = `
+variant: flatcar
+version: 1.0.0
+storage:
+  files:
+  - path: /etc/flatcar/nvidia-metadata
+    contents:
+      inline: |
+        {{ .NVIDIA_DRIVER_VERSION_LINE }}
+`
+
+const nvidiaOperatorTemplate = `
+variant: flatcar
+version: 1.0.0
+
+storage:
+  files:
+  - path: /etc/flatcar/nvidia-metadata
+    contents:
+      inline: |
+        NVIDIA_DRIVER_VERSION=570.86.15
```

Comment:
Should we upgrade the nvidia drivers in the coreos-overlay instead of hardcoding this version?

Reply:
R570 seems to be the first one that compiles correctly for aarch64 with kernel 6.6 and our gcc version, older ones have a build error. But according to nvidia docs (https://docs.nvidia.com/datacenter/tesla/drivers/index.html#cuda-and-drivers-table) R535 is actually LTS, so it makes for a better default for Flatcar. Nvidia driver's on amd64 are also much more widely used.

How about I upgrade only arm64 to 570 in coreos-overlay?

Reply:
> R570 seems to be the first one that compiles correctly for aarch64 with kernel 6.6 and our gcc version, older ones have a build error. But according to nvidia docs (https://docs.nvidia.com/datacenter/tesla/drivers/index.html#cuda-and-drivers-table) R535 is actually LTS, so it makes for a better default for Flatcar. Nvidia driver's on amd64 are also much more widely used.
> 
> How about I upgrade only arm64 to 570 in coreos-overlay?

Sounds good. amd64 will eventually catch up to arm64.

Another idea that came to my mind is to have a patch for R535 to fix the build issue, but I have no clue if it makes sense and how much effort that would be.